hail/python/hail/fs/hadoop_fs.py

```diff
@@ -60,12 +69,23 @@ def is_file(self, path: str) -> bool:
     def is_dir(self, path: str) -> bool:
         return self._jfs.isDir(path)
 
-    def stat(self, path: str) -> StatResult:
+    def fast_stat(self, path: str) -> FileStatus:
+        '''Get information about a path other than its file/directory status.
+
+        In the cloud, determining if a given path is a file, a directory, or both is expensive. This
+        method simply returns file metadata if there is a file at this path. If there is no file at
+        this path, this operation will fail. The presence or absence of a directory at this path
+        does not affect the return value of this method at all.
+
+        '''
+        raise NotImplementedError
```

Comment:
Implementing this operation in Java is a bit complicated and this PR already has many lines. Trust that I will follow up with an implementation.